src/solve.jl

```diff
@@ -62,6 +62,11 @@ function vectorized_solve(probs, prob::ODEProblem, alg;
     elseif alg isa GPUVern9
         kernel = vern9_kernel(dev)
     end
+
+    if dev isa CPU
+        @warn "Running the kernel on CPU"
```

Comment:
how could this happen?

Reply:
With KA.jl, we have defined overloads for `CPU` e.g. here https://github.com/SciML/DiffEqGPU.jl/blob/78e83aa708aae5c97b0490ad4380755a24b76b52/src/DiffEqGPU.jl#L74
I encountered this issue somehow with Julia 1.8, my backend wasn't configured properly, which led the adapt rules not be defined correctly, hence just returning the `prob` Array, even if I was doing `EnsembleGPUKernel(device)`. 
Adapt simply returns the Array, if the rules are defined.